scripts/Impurity-example.jl

```diff
@@ -0,0 +1,198 @@
+# to reflect various possible excitation options based on intuitions in physics.
+function init_dv(P,m,n)
+    aIni = BoseFS2C(BoseFS([n; zeros(Int, m-1)]), BoseFS([1; zeros(Int, m-1)]))
+    dv = InitiatorDVec(aIni=>1.0;style=IsDynamicSemistochastic())
+    empty!(dv)
+    c = (m+1)÷2 # finding the zero-momentum mode
+    if P == 0 # zero total momentum
+        bfs1 = zeros(Int, m);bfs2 = zeros(Int, m)
+        bfs1[c] = n; bfs2[c] = 1
+        dv[BoseFS2C(BoseFS(bfs1),BoseFS(bfs2))]+=1.0
+    elseif P == (n+1) # maximum total momentum
+        bfs1 = zeros(Int, m);bfs2 = zeros(Int, m)
+        bfs1[c+1] = n; bfs2[c+1] = 1
+        dv[BoseFS2C(BoseFS(bfs1),BoseFS(bfs2))]+=1.0
+    else
+        bfs1 = zeros(Int, m);bfs2 = zeros(Int, m)
+        bfs1[c] = n-(P-1); bfs1[c+1] = P-1; bfs2[c+1] = 1 # move impurity to c+1
+        dv[BoseFS2C(BoseFS(bfs1),BoseFS(bfs2))]+=1.0
+        
+        bfs1 = zeros(Int, m);bfs2 = zeros(Int, m)
+        bfs1[c] = n-P; bfs1[c+1] = P; bfs2[c] = 1 # move bosons to c+1
+        dv[BoseFS2C(BoseFS(bfs1),BoseFS(bfs2))]+=1.0
+        
+        bfs1 = zeros(Int, m);bfs2 = zeros(Int, m)
+        bfs1[c] = n; bfs2[c+P] = 1 # move impurity to c+P
+        dv[BoseFS2C(BoseFS(bfs1),BoseFS(bfs2))]+=1.0
+        
+        if (n-1) >= P >(n÷2)
+            bfs1 = zeros(Int, m);bfs2 = zeros(Int, m)
+            bfs1[c] = n-(P+1); bfs1[c+1] = P+1; bfs2[c-1] = 1 # move impurity to c-1 and a boson to c+1
+            dv[BoseFS2C(BoseFS(bfs1),BoseFS(bfs2))]+=1.0
+        end
+    end
+    return dv
+end
+
+# Note that the `dv` will be constructed with `InitiatorDVec()`, meaning 
+# that the initiator-FCIQMC algorithm will be used.
+
+# Now let's first do some MPI sanity checks and print some information:
+mpi_barrier() # optional, use for debugging and sanity checks
+@info "After barrier 1" mpi_rank() mpi_size() Threads.nthreads()
+
+# Now we specify parameters for constructing a two-component Hamiltonian
+P = 3 # total momentum
+m = 8 # number of modes
+na= 4 # number of non-impurity bosons
+γ = 0.2 # boson-boson coupling strength, dimensionless quantity
+η = 0.5 # impurity-boson coupling strength, dimensionless quantity
+T = m^2/2 # normalised hopping strength
+U = m*γ*na/(γ*na/(m*π^2) + 1) # converting γ to U
+V = m*η*na/(η*na/(m*π^2) + 1) # converting η to V
+# Here we use an initial address `aIni` for constructing the Hamiltonian, but 
+# it will not be used in the starting vector.
+aIni = BoseFS2C(BoseFS([na; zeros(Int, m-1)]), BoseFS([1; zeros(Int, m-1)]))
+ham = BoseHubbardMom1D2C(aIni;ta=T,tb=T,ua=U,v=V,dispersion=continuum_dispersion)
+
+
+
+# Now we can setup the Monte Carlo parameters
+steps_warmup = 10_000 # number of QMC steps running with a dummy Hamiltonian, see Stage 1
+steps_equilibrate = 10_000 # number of QMC steps running with the real `ham`
+steps_final = 10_000 # number of QMC steps running with G2 correlators, very slow, be caution!
```

Comment:
How slow is this script? The examples have been moved out of the testing job into the documentation job, but we don't want these examples to take too long...

Reply:
It takes about 1 min. I can make it quicker by using a smaller system or running fewer steps. What's your opinion for a ideal run time?